File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/05_hash_tables/ts/02_check_voter.ts
interface HashTable<T> {
    [key: string]: T;
}

const voted: HashTable<boolean> = {};

function check_voter(name: string): void {
    if (voted[name]) {
        console.log('kick them out!');
    } else {
        voted[name] = true;
        console.log('let them vote!');
    }
}

check_voter("tom"); // let them vote!
check_voter("mike"); // let them vote!
check_voter("mike"); // kick them out!

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/05_hash_tables/kotlin/02_check_voter.kt
val voted:HashMap<String,Boolean> = HashMap<String, Boolean>() 

fun checkVoter(name: String) {
 
    if(!voted.containsKey(name)){
        voted.put(name, true)
        println("let them vote!")
    } else {
        println("kick them out!")
    }
    
}

fun main(args: Array<String>) {
    
    checkVoter("tom")
    checkVoter("mike")
    checkVoter("mike")
}
